frontends/ol-search-ui/src/components/Searchbox.tsx

```diff
@@ -0,0 +1,191 @@
+import { ValidationError } from "ol-forms"
+import React, { useCallback, useState } from "react"
+import styled, { css } from "styled-components"
+
+const SearchboxContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+
+  height: 252px;
+  ${(props) => props.theme.media.phone} {
+    height: 115px;
+  }
+
+  padding: 0 20px;
+`
+
+const InputWrapper = styled.div`
+  position: relative;
+  height: 55px;
+  max-width: 500px;
+  width: 100%;
+`
+
+const SearchInput = styled.input`
+  &&& {
+    height: 100%;
+    padding-left: 50px;
+    font-size: 20px;
+    border-radius: 7px;
+    border: none;
+  }
+`
+
+const PositionedButton = styled.button<{ rightAlign?: boolean }>`
+  && {
+    background-color: rgba(0, 0, 0, 0);
+    color: ${({ theme }) => theme.color.fontGreyMid};
+    padding: 0px;
+    margin: 0px;
+    display: inline-flex;
+    align-items: center;
+  }
+  /* centered vertically */
+  position: absolute;
+  top: 50%;
+  transform: translateY(-50%);
+
+  ${({ rightAlign }) =>
+    rightAlign &&
+    css`
+      right: 10px;
+    `}
+  ${({ rightAlign }) =>
+    !rightAlign &&
+    css`
+      left: 10px;
+      width: 50px;
+    `}
+`
+
+const SearchboxIcon = styled.i<{ verticalOffset?: number }>`
+  font-size: 33px;
+  ${({ verticalOffset }) =>
+    verticalOffset &&
+    css`
+      transform: translateY(${verticalOffset}px);
+    `}
+`
+
+interface SearchSubmissionEvent {
+  target: {
+    value: string
+  }
+  /**
+   * Deprecated. course-search-utils calls unnecessarily.
+   */
+  preventDefault: () => void
+}
+type SearchSubmitHandler = (event: SearchSubmissionEvent) => void
+
+type SearchboxCommon = {
+  onSubmit: SearchSubmitHandler
+  validation?: string | null
+  autoFocus?: boolean
+  children?: React.ReactNode
+}
+
+export type SearchboxUncontrolled = {
+  onChange?: undefined
+  onClear?: undefined
+  value?: undefined
+} & SearchboxCommon
+export type SearchboxControlled = {
+  onChange: React.ChangeEventHandler
+  onClear: React.MouseEventHandler
+  value: string
+} & SearchboxCommon
+export type SearchboxProps = SearchboxUncontrolled | SearchboxControlled
+
+/**
+ * Renders a searchbox with Search and Clear icons, and optional child content.
+ * Can be used as a controlled or uncontrolled input.
+ */
+const Searchbox = (props: SearchboxProps): JSX.Element => {
```

Comment:
This was previously [CourseSearchbox](https://github.com/mitodl/open-discussions/blob/7973c4375bd7bc794da924aa4a0ddb1866e3d0a1/frontends/open-discussions/src/components/CourseSearchbox.js#L16)

I did make some logical changes within this component as I was trying to add types, but nothing major and it's all pretty thoroughly tested.

For example, on of the changes I made was using real buttons for the search and clear buttons. Before they were just icons. (Now icons wrapped in a button.)

This should be more accessible and easier for us... E.g., before the clear icon had:
```tsx
          <i
            className="material-icons clear-icon"
            onClick={onClear}
            onKeyPress={e => {
              if (e.key === "Enter" && onClear) {
                onClear()
              }
            }}
            tabIndex="0"
          >
            clear
          </i>
```
But with a real button you don't need separate `onKeyDown` and `onClick` handlers. Buttons can be clicked with the Enter key. (Or spacebar). So now it's just:
```tsx
          <PositionedButton onClick={onClear} rightAlign>
            <SearchboxIcon className="material-icons clear-icon">
              clear
            </SearchboxIcon>
          </PositionedButton>
```